improvements

andreasvc opened the pull request:
Summary of changes:
- Python 3 compatibility
- improve unicode handling
- implement finditer as generator
- support named groups in replacement strings
- support substitutions with count > 1
- re-organize code in several files

andreasvc committed:
implement finditer as generator

andreasvc committed:
Python 2/3 compatibility, support buffer objects, &c.

- Python 2/3 compatibility
- support searching in buffer objects (e.g., mmap)
- add module docstring
- some refactoring
- remove outdated Cython-generated file
- modify setup.py to cythonize as needed.

andreasvc committed:
minor changes

andreasvc committed:
re-organize code

andreasvc committed:
properly translate pos, endpos indices with unicode, &c.

- properly translate pos, endpos indices with unicode
- keep original unicode string in Match objects
- separate compile.pxi file

andreasvc committed:
new buffer API; precompute groups/spans; &c.

- use new buffer API
  NB: even though the old buffer interface is deprecated from Python 2.6,
  the new buffer interface is only supported on mmap starting from
  Python 3.
- avoid creating Match objects in findall()
- precompute groups and spans of Match objects, so that possibly encoded
  version of search string (bytestr / cstring) does not need to be kept.
- in _make_spans(), keep state for converting utf8 to unicode indices;
  so that there is no quadratic behavior on repeated invocations for
  different Match objects.
- release GIL in pattern_Replace / pattern_GlobalReplace
- prepare_pattern: loop over pattern as char *
- advertise Python 3 support in setup.py, remove python 2.5

andreasvc committed:
pickle Patterns; non-char buffers; &c.

- support pickling of Pattern objects
- support buffers from objects that do not support char buffer (e.g.,
  integer arrays); does not make a lot of sense, but this is what re does.
- enable benchmarks shown in readme by default; fix typo.
- fix typo in test_re.py

andreasvc committed:
handle named groups in replacement string; &c.

- handle named groups in replacement string
- store index of named groups in Pattern object instead of Match object.
- use bytearray for result in _subn_callback

andreasvc committed:
workaround for sub(...) with count > 1

andreasvc committed:
update performance table / missing features

andreasvc committed:
add flags parameter to toplevel functions

andreasvc committed:
Improve substitutions, Python 3 compatibility.

- when running under Python 3+, reject unicode patterns on
  bytes data, and vice versa, in according with general Python 3 behavior.
- improve Match.expand() implementation.
- The substitutions by RE2 behave differently from Python (character escapes,
  named groups, etc.), so use Match.expand() for anything but simple literal
  replacement strings.
- make groupindex of pattern objects public.
- add Pattern.fullmatch() method.
- use #define PY2 from setup.py instead of #ifdef hack.
- debug option for compilation.
- use data() instead of c_str() on C++ strings, and always supply length,
  so that strings with null characters are supported.
- bump minimum cython version due to use of bytearray typing
- adapt tests to Python 3; add b and u string prefixes where needed, &c.
- update README

andreasvc committed:
move functions around

andreasvc committed:
add count method

- add count method, equivalent to len(findall(...))
- use arrays in utf8indices
- tweak docstrings

axiak commented:
this looks good, I'm going to play around with this over the weekend

andreasvc committed:
document support syntax &c.

- add reference of supported syntax to main docstring
- add __all__ attribute defining public members
- add re's purge() function
- add tests for count method
- switch order of prepare_pattern() and _compile()
- rename prepare_pattern() to _prepare_pattern() to signal that it is
  semi-private

andreasvc committed:
add tests for bug with \\b

andreasvc committed:
fix Match repr

andreasvc committed:
Fix bugs; ensure memory is released; simplify C++ interfacing;

- Fix bug causing zero-length matches to be returned multiple times
- Use Latin 1 encoding with RE2 when unicode not requested
- Ensure memory is released:
  - put del calls in finally blocks
  - add missing del call for 'matches' array
- Remove Cython hacks for C++ that are no longer needed;
  use const keyword that has been supported for some time.
  Fixes Cython 0.24 compilation issue.
- Turn _re2.pxd into includes.pxi.
- remove some tests that are specific to internal Python modules _sre and sre

andreasvc committed:
add C++11 param; update URL

andreasvc committed:
fix setup.py unicode error

Commit:
Ignore non-matched groups when replacing with sub

From 3.5 onwards sub() and subn() now replace unmatched groups with
empty strings. See:

https://docs.python.org/3/whatsnew/3.5.html#re

This change removes the 'unmatched group' error which occurs when using
re2.

andreasvc committed:
Merge pull request #1 from pvaneynd/master

Ignore non-matched groups when replacing with sub

messense committed:
Fix groupdict decode bug

andreasvc committed:
Merge pull request #2 from messense/master

Fix groupdict decode bug

andreasvc committed:
disable non-matched group tests; irrelevant after dad49cd

andreasvc committed:
add -std=c++11 only for clang, because gcc on CentOS 6 does not support it

andreasvc committed:
update README

andreasvc commented:
Hi @axiak
It seems you're no longer maintaining pyre2; perhaps I could take over? I think this mostly involves handing over the pypi package. Let me know if you want to do this.
Cheers

andreasvc committed:
only translate unicode indices when needed

andreasvc committed:
use STL map for unicodeindices

podhmo committed:
macports support

andreasvc committed:
Merge pull request #3 from podhmo/macports

macports support

Commit:
Adding c++ 11 compile flag on Ubuntu

andreasvc committed:
Merge pull request #5 from mayk93/master

Adding c++ 11 compile flag on Ubuntu

andreasvc committed:
make -std=c++11 the default; fixes #4

andreasvc committed:
decode named groups even with bytes patterns; fixes #6

itdaniher commented:
I'd love to see think packaged in PyPi. Currently have a git URL in my requirements.txt to @andreasvc's branch.

andreasvc committed:
remove tests with re.LOCALE flag since it is not allowed with str in Python 3.6+

andreasvc committed:
disable failing test for known corner case

andreasvc committed:
Disable dubious tests

- All tests pass.
- Don't test for exotic/deprecated stuff such as non-initial flags in
  patterns and octal escapes without leading 0 or triple digits.
- Known corner cases no longer reported as failed tests.
- support \b inside character class to mean backspace
- use re.error instead of defining subclass RegexError; ensures that
  exceptions can be caught both in re2 and in a potential fallback to re.

andreasvc committed:
Add contains() method

- contains() works like match() but returns a bool to avoid creating a
  Match object. see #12
- add wrapper for re.Pattern so that contains() and count() methods are
  also available when falling back to re.

yoav-orca committed:
created pyproject.toml

Poetry and other modern build system need to know which build-tools to
install prior to calling setup.py. added a pyproject.toml to specify
cython as a dependency.

yoav-orca committed:
Creating github actions for building wheels

andreasvc committed:
Merge pull request #14 from yoav-orca/master

Support building wheels automatically using github actions

andreasvc committed:
bump version

andreasvc committed:
change package name for pypi

andreasvc committed:
bump version

Commit:
fix pickle_test (tests.test_re.ReTests) ... ERROR (run tests with nose)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: pkg: workaroud for manylinux dependency install error, add release flow

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #18 from freepn/github-ci

workaroud for manylinux dependency install error plus release automation

andreasvc committed:
bump version

Commit:
new: pkg: convert to pep517 with cmake and pybind11 build config

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: ci: update wheel builds for Linux, Macos, and Windows

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: pkg: update macos wheel repair, simplify wheel building

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: ci: make sure wheel path is correct for uploading

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: doc: add .gitchangelog.rc and generated CHANGELOG.rst (keep HISTORY)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #20 from freepn/new-bld

New cmake and pybind11 build setup

andreasvc committed:
update README.rst. fixes #21

andreasvc committed:
bump version

Commit:
add missing tests to sdist package, update readme and ci worflows (#1)

readme: update badges, merge install sections, fix some rendering issues

Commit:
update changelog (and trigger ci rebuild)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: pkg: add simplejson to test deps, remove excelude for wikidata blob

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: test: rename imported test helpers to avoid any discovery issues

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #22 from sarnold/missing-tests

add missing tests to sdist package, update readme and ci worflows (#1)

andreasvc committed:
update README, fix Makefile

andreasvc committed:
Makefile: default to Python 3

andreasvc committed:
tweak order of badges

andreasvc committed:
use pytest; fixes #23

andreasvc committed:
fix Python 2 compatibility

Commit:
fix: dev: encode test data, add pytest and tox cfgs, update reqs

* also some readme and manifest cleanup (fix title, add global exclude)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: dev: add more clean plus cleanup cfgs and remove cruft

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: test: handle invalid escape sequence warnings, revert path changes

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: ci: update workflows and tox cfg (use tox for smoke test)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: pkg: add .gitattributes and revert conda test change

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: pkg: add pytest to conda recipe (plus test patch)

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #24 from sarnold/tst-cleanup

Test cleanup

andreasvc committed:
fix narrow unicode detection

andreasvc committed:
fix "make test"; rename doctest files for autodetection

andreasvc committed:
fix conda patch

andreasvc committed:
fix fix of conda patch

andreasvc committed:
another one

Commit:
fix: test: apply test patch, cleanup tox and pytest args

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
chg: pkg: remove test patch from conda recipe

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #25 from sarnold/last-moves

Last moves

andreasvc committed:
fix "make test" and "make test2"

andreasvc committed:
fix infinite loop on substitutions of empty matches; fixes #26

andreasvc committed:
add test for #26

andreasvc committed:
improve fix for #26

andreasvc committed:
bump version

Commit:
move pypi upload to end of release.yml, use gitchangelog action

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #28 from sarnold/use-action

Workflow cleanup

andreasvc committed:
bump version again

Commit:
add missing sdist job and artifact check to workflows, bump version

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #30 from sarnold/release-pr

workflow updates

JustAnotherArchivist committed:
Make Match objects subscriptable

Fixes #31

JustAnotherArchivist committed:
Add test for Match subscripting

andreasvc committed:
Merge pull request #32 from JustAnotherArchivist/match-getitem

Make Match objects subscriptable

Commit:
fix: dev: add conda-only patch for test_emptygroups failure

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: dev: conda build workflow: drop py 3.6 and add py 3.10

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: dev: limit conda build workflow to py 3.8 and 3.9

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: apply emptygroups fix and remove conda-only patch, also

* release workflow: restrict pypi upload to repo owner
* tox.ini: replace deprecated pep517 module, update deploy url

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #33 from sarnold/conda-patch

Conda patch for None vs empty string change

andreasvc committed:
remove python versions for make valgrind

andreasvc committed:
add NOFLAGS and RegexFlags constants; #41

andreasvc committed:
make tests pass on my system; if this behavior turns out to be inconsistent across versions/platforms, maybe the test should be disabled altogether. #27

andreasvc committed:
document lack of support for possessive quantifiers and atomic groups

andreasvc committed:
support fallback to Python re for possessive quantifiers

andreasvc committed:
include current notification level in cache key

this prevents a cached regular expression being used that was created
with a different notification level.

For example, the following now generates the expected warning:

    In [1]: import re2
    In [2]: re2.compile('a*+')
    Out[2]: re.compile('a*+')
    In [3]: re2.set_fallback_notification(re2.FALLBACK_WARNING)
    In [4]: re2.compile('a*+')
    <ipython-input-5-041122e221c7>:1: UserWarning: WARNING: Using re module. Reason: bad repetition operator: *+
      re2.compile('a*+')
    Out[4]: re.compile('a*+')

andreasvc committed:
fix #42

sarnold committed:
chg: dev: update python, deps, GH action versions, and tox file

* update pybind11 usage and set cmake python vars to Title_CASE
* refactor cmake extension build to use pybind11 module bits
* move emptygroups test from "differences"

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
fix: dev: remove failing subscript test in single match group

* cleanup asserts and add groups() test

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: update packaging files, add setuptools_scm support

* refactor setup.py after pybind11 upstream changes

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: bump cibw version, update workflows and min py version

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: add/update cfgs, modernize cibw build workflow

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: update cmake and workflow files, delete unused bits

* cleanup ci workflow, remove crufty makefile with deprecated
  setup.py commands
* remove the package_dir bit from setup.py

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: cleanup metadata and test imports, disable platform whl tests

* check if find_package py3 works across all CI runners

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: enable findpython policy, use	matrix uploads

* no epel pkgs for linux aarch64, enable PYBIND11_FINDPYTHON
* set macos deployment target to 10.9

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

sarnold committed:
chg: dev: try pkgconf on windows CI runner

Signed-off-by: Steve Arnold <sarnold@vctlabs.com>

Commit:
fix: dev: bump CMake to c++17 with extensions to build against re2-0.2024.07.02

Signed-off-by: Stephen L Arnold <nerdboy@gentoo.org>

Commit:
fix: dev: bump ubuntu and mac workflow runners

* revert to macos-13 with the same version as target
* In Theory this should get us full c++17

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
chg: swap out flake8 for cython-lint, update setup files, remove pep8 cfg

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
fix: cleanup tests and fix a raw string test, enable more win32

* split all runners into separate arch via matrix
* macos does need macos-14 to get a proper arm64 build

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
chg: switch conda workflow to condadev environment

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
new: doc: add basic sphinx docs build using apidoc, update changelog

* update .gitignore and .gitchangelog.rc and (re)generate new changelog
* add sphinx docs build using apidoc extension and readme/changelog
  symlinks
* rst apidoc modules are auto-generated and are in .gitignore
  along with the generated html dir
* add dependencies to packaging and add docs/changes cmds to tox file.
  Includes a tox extension for shared tox environments; the new tox
  commands are an example of this => 4 cmds using one tox env

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
chg: dev: make sure conda-devenv installs specific pkgs with pip

* this is essentially a workaround for non-pypi pkg cruft

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
chg: dev: switch conda env to use mamba with newer workflow cmds

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
chg: dev: one more conda-devenv refactor based on latest docs

* also cleanup the wheel artifact check, download to artifacts/

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #47 from sarnold/py-cleanup

CI, python, and cmake cleanup

Commit:
chg: add gh workflow for sphinx build/deploy

* cleanup docs config, remove dicey sphinx_git extension
* switch readme badge, download wheel artifacts to single directory

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
fix: dev: update release workflow for new platform wheels

* also cleanup sphinx workflow

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

Commit:
chg: dev: minor workflow changes to kickstart deployment

Signed-off-by: Stephen Arnold <nerdboy@gentoo.org>

andreasvc committed:
Merge pull request #50 from sarnold/sphinx-apidocs

add basic sphinx docs build using apidoc

otetard committed:
feat: Add support for Python 3.13

otetard committed:
refactor: Drop support for Python 2

otetard committed:
fix: Don’t use cmake 4 to build

otetard committed:
ci: Fix “release” workflow

andreasvc committed:
Merge pull request #51 from otetard/python313

- Drop support for Python 2.
- Add support for Python 3.13 in tests & ciwheelbuild, and move tests/test_charliterals.txt to unitest as the previous test (Python 3.13 error changed).
- Don’t use cmake 4.0 to build the library.
- Fix GitHub Actions’ “release” workflow.

andreasvc committed:
disable linkcheck

adepasquale committed:
Add set_fallback_module() to give users choice (#55)

* Add set_fallback_module() to give users choice

Some patterns are not supported by re2.

The existing set_fallback_notification() allows the user to decide what
to do when a pattern is not supported: be quiet, emit a warning, raise
an exception.

This commits adds set_fallback_module() to allow the user to choose
which module use as a fallback. The default/initial choice is Python's
re module, to be consistent with current behavior.

The change has been tested using regex as a fallback module, which in
some scenarios might still be faster than Python's re.

andreasvc committed:
fix gh-action-pypi-publish version

PLPeeters committed:
Fix FallbackPattern.__reduce__ (#58)

andreasvc committed:
Bump version, support Py3.10-3.14, drop Mac X86_64, Win 32bit

andreasvc committed:
Merge branch 'master' of github.com:andreasvc/pyre2

andreasvc committed:
use macos-14 instead of latest

andreasvc committed:
drop macos for conda build

andreasvc committed:
bump version

andreasvc committed:
attempt at fixing pypi uploads

andreasvc committed:
Update README.rst

andreasvc committed:
Merge branch 'master' of github.com:andreasvc/pyre2

andreasvc committed:
address issues with null bytes

andreasvc committed:
fix null test

andreasvc committed:
fix null test (again)

andreasvc committed:
fix conda workflow

andreasvc committed:
another atttempt at fixing conda workflow

andreasvc committed:
attempt at diagnosing conda workflow issue

andreasvc committed:
one more attempt at fixing conda workflow

blemoallic committed:
fix: use an LRU cache for cached patterns (#60)
